geom/FieldPointerUtilities.F90

```diff
@@ -859,4 +899,51 @@ subroutine GetFieldsUndef_r8(fields,undef_values,rc)
      _RETURN(_SUCCESS)
   end subroutine GetFieldsUndef_r8
 
+subroutine Destroy(Field,RC)
+    type(ESMF_Field),          intent(INOUT) :: Field
+    integer, optional,         intent(OUT  ) :: RC
+
+    integer                               :: STATUS
+
+    real(kind=ESMF_KIND_R4), pointer      :: VR4_1D(:), VR4_2D(:,:), VR4_3D(:,:,:), VR4_4D(:,:,:,:)
+    real(kind=ESMF_KIND_R8), pointer      :: VR8_1D(:), VR8_2D(:,:), VR8_3D(:,:,:), VR8_4D(:,:,:,:)
+    integer                      :: rank
+    type(ESMF_TypeKind_Flag)     :: tk
+    logical :: esmf_allocated
+
+    call ESMF_FieldGet(Field,typekind=tk,dimCount=rank,isESMFAllocated=esmf_allocated,_RC)
+    if (.not. esmf_allocated) then
+       if (tk == ESMF_TYPEKIND_R4 .and. rank == 1) then
+          call ESMF_FieldGet(Field,0,VR4_1d,_RC)
+          deallocate(VR4_1d,_STAT)
+       else if (tk == ESMF_TYPEKIND_R8 .and. rank == 1) then
+          call ESMF_FieldGet(Field,0,VR8_1d,_RC)
+          deallocate(VR8_1d,_STAT)
```

Comment:
I'm tempted to ask you to try using the existing utilities to get a 1D pointer and deallocate that.   Not sure what the standard says about that though, so even if it works might be unsafe.

Reply:
MAPL3 will almost always let ESMF do the allocation, so this block can maybe be removed then.